search-app/docs/api.md

```diff
@@ -244,8 +245,47 @@ The CMR contains many more results than can be returned in a single response so
 
 You can not page past the 1 millionth item. Please contact the CMR Team through the [CMR Client Developer Forum](https://wiki.earthdata.nasa.gov/display/CMR/CMR+Client+Developer+Forum) if you need to retrieve items in excess of 1 million from the CMR. Additionally granule queries which do not target a set of collections are limited to paging up to the 10000th item.
 
+#### <a name="search-after"></a> Search After
+
+Search After supersedes scrolling. Search After allows the retrieval of all results of a query in an efficient manner. It is supported through the `CMR-Search-After` header. Search After is primarily intended to support harvesting of metadata.
+
+Search After is only supported for parameter queries and JSON queries. All query parameters are available with the exception of the `page_num` and `offset` parameters.
+
+Search After is stateless, it is always resolved against the latest version of the data. Any search against CMR that has results not fully returned in the current request will return a `search-after` value in the `CMR-Search-After` header of the search response. User can then pass this returned value in the `CMR-Search-After` header of the following request to retrieve the next page of result based on the specified page_size. Each search request will result in a new `search-after` value returned in the `CMR-Search-After` response header. Supplying the new `search-after` value in the following request's `CMR-Search-After` header will retrieve the next page. The `CMR-Hits` header is useful for determining the number of requests that will be needed to retrieve all the available results.
+
+When all the results have been returned, the subsequent search will return an empty result set and no `CMR-Search-After` header in the response.
+
+Different from scrolling requests, each search-after request needs to supply all the search parameters, and the `CMR-Search-After` header needs to be updated with the new value returned from the previous search to page through the whole result set. Although user can change the search parameters and still get results back as long as the sort order of the search is unchanged, it breaks the rationale of paging and offers no real use case. Thus user should always supply the same search parameters while using Search After requests to page through a large result set.
```

Comment:
One of the features we have been taking advantage of is not having to know the query that passed to CMR to initiate a scroll session. We have a different application initiating the scroll session than the one that consumes the results. Having to pass the same query in can be difficult (especially with a large POST body or with a shapefile).

Can you think of a way of supporting Search-After without having to pass in the query again?

Reply:
This would be completely against the stateless nature of ES search-after feature. I would like to keep the CMR search-after feature as is to utilize the native ES search-after feature. What you are asking for can be done, but I would like to write a separate ticket to implement (maybe) a new endpoint for the session based search.